### PR TITLE
Proposal to add ZodiPy to affiliated packages (pyOpenSci/software-submission#161)

### PR DESCRIPTION
**Pull request by MetinSa**
EDIT: Also submitted https://github.com/pyOpenSci/software-submission/issues/161

ZodiPy is a new Python tool for modelling the zodiacal emission seen by an arbitrary Solar System observer, which can be used for removal of both thermal emission and scattered sunlight from interplanetary dust in astrophysical data. One of our main goals with ZodiPy is to make zodiacal emission simulations more accessible by providing the community with a simple Python interface to existing models. We recently published a [paper on ZodiPy](https://arxiv.org/abs/2205.12962) (which has also been accepted to A&A) where we describe in more detail modelling of zodiacal emission and the approach taken by ZodiPy.

ZodiPy uses Astropy both in the public API and internally. When using one of the methods provided to simulate the zodiacal emission, the user needs to use `Quantity` and `Time` types in input arguments such as frequency/wavelength, angles on the sky, and the time of observation. Internally, the `Time` object is used to compute the heliocentric ecliptic position of the Earth and the observer with the `SkyCoord` and other functionality in `astropy.coordinates`.

### CODE REVIEW

File: affiliated/registry.json [hunk 1]
```diff
@@ -1,5 +1,23 @@
 {
     "packages": [
+        {
+            "name": "ZodiPy",
+            "maintainer": "Metin San <metin.san@astro.uio.no>",
+            "stable": true,
+            "home_url": "https://cosmoglobe.github.io/zodipy/",
+            "repo_url": "https://github.com/cosmoglobe/zodipy",
+            "pypi_name": "zodipy",
+            "description": "ZodiPy is a Python tool for simulating the zodiacal emission that a solar system observer sees, either in the form of timestreams or binned HEALPix maps.",
```
Comment (dhomeier) [inline affiliated/registry.json:10]:
```suggestion
            "description": "ZodiPy is a Python tool for simulating the zodiacal emission that a solar system observer sees, either in the form of timestreams or binned HEALPix maps.",
            "coordinated": false,
```
This is a required field that we have forgotten to add to the template yet.

**Comment by dhomeier** [comment]
Thank you for proposing this package as an affiliated package! I'm happy to confirm that your package is now under review, and we will post the results of the review here and on the mailing list.

**Comment by dhomeier** [comment]
@MetinSa: apologies for leaving your proposal in a dormant state for so long!
In part this was due to a transition in our review system we were making over the last year, and which took also longer than foreseen to complete.
We now expect to have a review for your package within the next week or two.
But since we are at this point, I would also like to some details of the migration of our Affiliated Package review process to a new system in partnership with pyOpenSci now described in [APE 22](https://github.com/astropy/astropy-APEs/blob/main/APE22.rst#in-a-nutshell).

What this means for all packages such as yours, which have applied for Affiliated status before APE 22 was passed is, that we still offer you to continue the package review in the previous mode, but also want to give you the option to complete this process through the pyOpenSci page.

In that case we would have to find a second reviewer, but the (open) pyOpenSci review is generally expected to take not more than 3 weeks, and reviews and responses will be openly discussed as pyOpenSci GitHub issues as described in their [submission guidelines](https://github.com/pyOpenSci/software-submission?tab=readme-ov-file#how-to-submit-a-package-for-review). Your package would also be checked for satisfying the [pyOpenSci guidelines](https://www.pyopensci.org/software-peer-review/how-to/author-guide.html) in addition to Astropy-specific requirements such as integration into the Astropy ecosystem, but largely the criteria will be what we also expect for the Astropy Affiliated packages.
As a benefit this would result in
1. additional listing as a [pyOpenSci package](https://www.pyopensci.org/python-packages.html#our-accepted-python-open-source-packages)
2. automatic submission for publication in [JOSS](https://www.pyopensci.org/software-peer-review/partners/joss.html)

Please comment here on your preferences, or if you need additional information on this process.
Thanks!

**Comment by dhomeier** [comment]
@MetinSa I am happy to tell you that this package has now been reviewed for inclusion in the Astropy affiliated package
ecosystem by a member of the Astropy community, and I have synthesised the results of the review here.

You can find out more about our review criteria in [Reviewing affiliated packages](https://github.com/astropy/astropy-procedures/blob/main/documents/affiliated_package_review_guidelines.md).
For each of the review categories below we have listed the score and have
included some comments when the score is not green.

<table>
<tr><td><b>Functionality/Scope</b></td><td>
<a href="https://github.com/astropy/astropy-procedures/blob/main/documents/affiliated_package_review_guidelines.md#functionality-functionality">
Specialized package
</a></td></tr>

<tr><td colspan=2><i>ZodiPy looks like a very healthy package. Highlighted below are some spots where there is room for improvement.</i></td></tr>

<tr><td><b>Integration with Astropy ecosystem</b></td><td>
<a href="https://github.com/astropy/astropy-procedures/blob/main/documents/affiliated_package_review_guidelines.md#integration-with-astropy-ecosystem--ecointegration">
Orange
</a></td></tr>

<tr><td colspan=2><i>It could have much more integration with Astropy core and Astropy affiliated packages. Aside from Astropy `units`, the public API does not have much integration with Astropy. Here are some issues:

- The `get_emission_ang` method basically depends on two coordinates that define the ray from the observer to the direction in which they are looking (plus a time and a wavelength). The vertex of the ray can be specified in Cartesian coordinates and the direction of the ray in polar coordinates. Internally, the method instantiates `SkyCoord` objects in order to do positional astronomy calculations. It would be ideal if the user could specify both coordinates using `SkyCoord` instances. That would unify 5 arguments (`theta`, `phi`, `obs`, `obs_pos`, `lonlat`) into only two.
- Conversion between coordinate frames is done using `healpy.Rotator` but could be done more simply and idiomatically with `SkyCoord` frame transformations.
- There is a synthetic photometry feature to integrate the observed flux over a bandpass. ZodiPy could employ the Astropy affiliated package `synphot` here.
- ZodiPy can evaluate the emission on `HEALPix` maps, but uses `healpy` to do this. It could use the Astropy affiliated package `astropy-healpix` here.
- ZodiPy can plot emission components on all-sky maps, but uses `healpy` to do this. It could use `reproject` +  
   `astropy-healpix` here for more flexible plotting.</i></td></tr>

<tr><td><b>Documentation</b></td><td>
<a href="https://github.com/astropy/astropy-procedures/blob/main/documents/affiliated_package_review_guidelines.md#documentation-documentation">
Green
</a></td></tr>

<tr><td colspan=2><i>
</i></td></tr>

<tr><td><b>Testing</b></td><td>
<a href="https://github.com/astropy/astropy-procedures/blob/main/documents/affiliated_package_review_guidelines.md#testing-testing">
Green
</a></td></tr>


<tr><td><b>Development status</b></td><td>
<a href="https://github.com/astropy/astropy-procedures/blob/main/documents/affiliated_package_review_guidelines.md#development-status-devstatus">
Green
</a></td></tr>

<tr><td colspan=2><i> Some additional notes and suggestions from the reviewer follow:

The approaches to Numpy vectorization and Python multiprocessing parallelism are somewhat fixed function, and may not be performant for all use cases.

- Methods do not support Numpy broadcasting for arbitrary input shapes. For example, it appears that `obs_time` must be scalar. For another, this line in `get_obs_and_earth_positions` discards the shape of the input arguments:
```python
   return obs_position.reshape(3, 1, 1).value, earth_position.reshape(3, 1, 1).value
```
- ZodiPy supports parallelism using Python multiprocessing, but it creates its own process pool and discards it when it is done. It would be nice if it supported the pool being passed by the user so that the cost of spawning child Python interpreters is not borne repeatedly. Also there are situations where it is necessary to use a custom Pool subclass (e.g. [billiard](https://pypi.org/project/billiard/)).</i></td></tr>

<tr><td><b>Python 3 compatibility</b></td><td>
<a href="https://github.com/astropy/astropy-procedures/blob/main/documents/affiliated_package_review_guidelines.md#python-3-compatibility-python3">
Green
</a></td></tr>

</table>

**Summary/Decision**: With the overall positive review above your package already meets our legacy criteria for Astropy Affiliated packages, so I can offer you at this point to list your package on https://www.astropy.org/affiliated/.
I still encourage you to proceed with the new [pyOpenSci application process](https://github.com/astropy/astropy.github.com/pull/495#issuecomment-1949180178) by submitting this application as is, following https://www.pyopensci.org/software-peer-review/our-process/review-timeline.html#author-submits-their-package-for-review. At that point we would already have one reviewer, and as soon as a second reviewer has completed their report (and potential comments been addressed), your package would be listed with pyOpenSci and JOSS as well.

**Comment by pllim** [comment]
@dhomeier , does this mean you want to add ZodiPy to the "legacy" listing as well, before I freeze it over at #573 ?

**Review by dhomeier — APPROVED** [review]
Yes, this can go into the _legacy_ section as there is no further action required on the Astropy side.

File: affiliated/registry.json [hunk 2]
```diff
@@ -1,6 +1,24 @@
...
+            "review": {
+               "functionality": "Specialized package",
+               "ecointegration": "Partial",
+               "documentation": "Good",
+               "testing": "Good",
+               "devstatus": "Good",
+               "python3": "Yes",
+               "last-updated": "2024-02-24"
+            }
+          },
```
Comment (dhomeier) [inline affiliated/registry.json:21]:
```suggestion
          },
          {
```

**Comment by pllim** [comment]
Please remember to use "squash and merge". Thanks!

**Pull request merged**